src/CodeCompliance/Domain/AbstractCodeComplianceCheck.php

```diff
@@ -27,12 +28,19 @@ abstract class AbstractCodeComplianceCheck implements CodeComplianceCheckInterfa
      */
     protected FilterService $filterService;
 
+    /**
+     * @var \CodeCompliance\Domain\Adapter\CodeBaseReaderInterface
+     */
+    protected CodeBaseReaderInterface $codeBaseAdapter;
```

Comment:
Can it be a service?

Reply:
renamed to CodeBaseServiceInterface

Reply:
As to me it should be a separate feature.